daemons/attrd/attrd_messages.c

```diff
@@ -90,6 +90,25 @@ handle_refresh_request(pcmk__request_t *request)
     }
 }
 
+static xmlNode *
+handle_update_request(pcmk__request_t *request)
+{
+    if (request->peer != NULL) {
+        const char *host = crm_element_value(request->xml, PCMK__XA_ATTR_NODE_NAME);
+        crm_node_t *peer = crm_get_peer(0, request->peer);
+
+        attrd_peer_update(peer, request->xml, host, false);
+        pcmk__set_result(&request->result, CRM_EX_OK, PCMK_EXEC_DONE, NULL);
+        return NULL;
+    } else {
+        /* Because attrd_client_update can be called recursively, we send the ACK
+         * here to ensure that the client only ever receives one.
+         */
+        attrd_send_ack(request->ipc_client, request->ipc_id, request->flags);
+        return attrd_client_update(request);
```

Comment:
Another possibility would be to add an ack_needed argument to attrd_client_update(). Not necessary, just thinking out loud